### PR TITLE
DBZ-577 Numeric arrays correctly working

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-577

### CODE REVIEW

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java [hunk 1]
```diff
@@ -127,7 +127,7 @@ public SchemaBuilder schemaBuilder(Column column) {
             case PgOid.BPCHAR_ARRAY:
                 return SchemaBuilder.array(SchemaBuilder.OPTIONAL_STRING_SCHEMA);
             case PgOid.NUMERIC_ARRAY:
-                return SchemaBuilder.array(numericSchema(column).optional());
+                return SchemaBuilder.array(numericSchema(column).optional().build());
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java]: Wow, quite a poor API design which allows for this kind of bug :(

File: debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java [hunk 2]
```diff
@@ -278,6 +278,8 @@ private ValueConverter createArrayConverter(Column column, Field fieldDefn) {
                 .jdbcType(schema.columnTypeNameToJdbcTypeId(PostgresSchema.parse(elementTypeName).table()))
                 .type(elementTypeName)
                 .optional(true)
+                .scale(column.scale())
```
Comment (gunnarmorling) [inline debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java]: Should we check for the value to be not -1?

Comment (jpechane) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java]: `-1` is a valid value for varibale scaled decimal, so this is intentional

Comment (gunnarmorling) [reply debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java]: Ok, seems legit. Btw. in the constructor of `ColumnImpl` there're two asserts rejecting anything < -1. That's potentially not correct then if it's a variable scale decimal column?

**Comment by gunnarmorling** [comment]
Applied. Thanks, @jpechane!

**Pull request merged**